serverFiles/models/articles.models.js

```diff
@@ -44,12 +44,32 @@ function selectCommentsByArticleId(article_id) {
     )
     .then(({ rows }) => {
       if (!rows.length) {
-        return checkArticleExists();
+        return checkArticleExists(article_id);
       }
       return rows;
     });
 }
 
+function changeVotesByArticleId(article_id, newVoteData) {
+  const regexPattern = /\d/g;
+  const isNumber = regexPattern.test(article_id);
+  if (!isNumber) {
+    return Promise.reject({ status: 400, msg: 'Bad request' });
+  }
```

Comment:
A valid approach, but this could also be dealt with by catching the resulting PSQL error in your handlePSQL error handling function, as handing PSQL queries the wrong data type makes them throw an error, with a certain code.